<h1>ACTIVITY 1 - PALMER PENGUINS DATA ANALYTICS PROJECT</h1>
<hr>
<h3>Analyst : Ralph Richmond D. Amarillo </h3>

In [1]:
#Import Libraries
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings("ignore")

<h3>Coverting Dataset(s) into DataFrame</h3>

<h3>Requirements</h3>
<ul>
    <li> Name of the Dataframe</li>
    <li> File path</li>
</ul>

In [2]:
penguins = pd.read_csv('datasets\\penguins.csv')

In [3]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

<h3>Descriptive Analytics</h3>

In [4]:
penguins.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

In [5]:
penguins.describe().T

count         mean         std     min       25%      50%  \
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                      75%     max  
bill_length_mm       48.5    59.6  
bill_depth_mm        18.7    21.5  
flipper_length_mm   213.0   231.0  
body_mass_g        4750.0  6300.0  
year               2009.0  2009.0

<h1>Checking for Null Values</h1>

In [6]:
penguins.isnull()

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0      False   False           False          False              False   
1      False   False           False          False              False   
2      False   False           False          False              False   
3      False   False            True           True               True   
4      False   False           False          False              False   
..       ...     ...             ...            ...                ...   
339    False   False           False          False              False   
340    False   False           False          False              False   
341    False   False           False          False              False   
342    False   False           False          False              False   
343    False   False           False          False              False   

     body_mass_g    sex   year  
0          False  False  False  
1          False  False  False  
2          False  False  False  
3           True   True  False  
4          False  False  False  
..           ...    ...    ...  
339        False  False  False  
340        False  False  False  
341        False  False  False  
342        False  False  False  
343        False  False  False  

[344 rows x 8 columns]

In [7]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<h3>Filtering a column</h3>

In [8]:
penguins['species']

0         Adelie
1         Adelie
2         Adelie
3         Adelie
4         Adelie
         ...    
339    Chinstrap
340    Chinstrap
341    Chinstrap
342    Chinstrap
343    Chinstrap
Name: species, Length: 344, dtype: object

In [9]:
penguins['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

In [10]:
penguins['species'] == 'Chinstrap'

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340     True
341     True
342     True
343     True
Name: species, Length: 344, dtype: bool

<h3>Filtering Data</h3>

In [11]:
adelie_DF = penguins[penguins['species'] == "Adelie"]
adelie_DF

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
149  Adelie      Dream            37.8           18.1              193.0   
150  Adelie      Dream            36.0           17.1              187.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
147       3475.0  female  2009  
148       3450.0  female  2009  
149       3750.0    male  2009  
150       3700.0  female  2009  
151       4000.0    male  2009  

[152 rows x 8 columns]

In [12]:
 gentoo_df = penguins[penguins['species'] == 'Gentoo']

In [13]:
gentoo_df

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe            46.1           13.2              211.0   
153  Gentoo  Biscoe            50.0           16.3              230.0   
154  Gentoo  Biscoe            48.7           14.1              210.0   
155  Gentoo  Biscoe            50.0           15.2              218.0   
156  Gentoo  Biscoe            47.6           14.5              215.0   
..      ...     ...             ...            ...                ...   
271  Gentoo  Biscoe             NaN            NaN                NaN   
272  Gentoo  Biscoe            46.8           14.3              215.0   
273  Gentoo  Biscoe            50.4           15.7              222.0   
274  Gentoo  Biscoe            45.2           14.8              212.0   
275  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  
152       4500.0  female  2007  
153       5700.0    male  2007  
154       4450.0  female  2007  
155       5700.0    male  2007  
156       5400.0    male  2007  
..           ...     ...   ...  
271          NaN     NaN  2009  
272       4850.0  female  2009  
273       5750.0    male  2009  
274       5200.0  female  2009  
275       5400.0    male  2009  

[124 rows x 8 columns]

In [14]:
(penguins['species'] =='Chinstrap') & (penguins['flipper_length_mm'] >= 18)

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340     True
341     True
342     True
343     True
Length: 344, dtype: bool

<h3>Checking Null Values</h3>

In [15]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<h3>Changing the Data type</h3>

In [16]:
penguins['species'] = penguins['species'].astype('category')
penguins['year'] = penguins['species'].astype('category')

In [17]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    object  
 2   bill_length_mm     342 non-null    float64 
 3   bill_depth_mm      342 non-null    float64 
 4   flipper_length_mm  342 non-null    float64 
 5   body_mass_g        342 non-null    float64 
 6   sex                333 non-null    object  
 7   year               344 non-null    category
dtypes: category(2), float64(4), object(2)
memory usage: 17.2+ KB


<h1>CORRELATION ANALYSIS</h1>
<hr>
<h3>Testing Correlation usinng Pearson R</h3>
<hr>
<strong>Pearsonn R</strong> can test correlation between two Variable, given that both Variable are <strong>NUMERICAL DATA</strong>

In [18]:
flipper_mass = penguins['body_mass_g'].corr(penguins['flipper_length_mm'])

In [19]:
flipper_mass

np.float64(0.8712017673060113)

<h1>Testing Correlation Between Numerical and Catergorical Data</h1>
<hr>
<h3>Body Mass V Sex</h3>

In [20]:
#import scipy.stats
from scipy.stats import f_oneway

In [21]:
#fileter data
penguins[penguins['sex'] == 'male']['body_mass_g']

0      3750.0
5      3650.0
7      4675.0
13     3800.0
14     4400.0
        ...  
334    3800.0
336    3950.0
339    4000.0
341    3775.0
342    4100.0
Name: body_mass_g, Length: 168, dtype: float64

In [22]:
penguins[penguins['sex'] == 'famale']['body_mass_g']

Series([], Name: body_mass_g, dtype: float64)

In [23]:
no_use, p_value = f_oneway(penguins[penguins['sex'] == 'male']['body_mass_g'], penguins[penguins['sex'] == 'female']['body_mass_g'])

In [24]:
p_value

np.float64(4.897246751596225e-16)

<h1>FELEC3B-ACTIVITIES</h1>
<hr>
<h3>Question : 1</h3>
<h3>Which island has the heaviest average body mass of penguins?<h3>

In [25]:
years = [2007, 2008, 2009]

In [26]:
for y in years:
    gentoo_avg = penguins[(penguins['species'] == 'Gentoo') & (penguins['year'] == y)]['bill_length_mm'].mean()
    print(y, ":", gentoo_avg)


2007 : nan
2008 : nan
2009 : nan


<h3>Insight : Gentoo males have the longest bill length, with the highest recorded measurement being 59.6 mm.</h3>

<hr>

<h3>Question : 2</h3>
<h3>What sex have the shortest bill length?</h3>

In [27]:
 penguins['bill_length_mm'].min()

np.float64(32.1)

In [28]:
penguins[penguins['bill_length_mm'] == 32.1]

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
142  Adelie  Dream            32.1           15.5              188.0   

     body_mass_g     sex    year  
142       3050.0  female  Adelie

<h3>Insight : Based on this, Adelie females have the shortest bill length, with the minimum recorded being 32.1 mm</h3>

<hr>

<h3>Question : 3 </h3>
<h3>How many penguins were documented in each year?</h3>

In [29]:
penguins['year'].value_counts().sort_index()

year
Adelie       152
Chinstrap     68
Gentoo       124
Name: count, dtype: int64

<h3>Insight : We can see that Adelie penguins are the most common who have 152, while Chinstrap penguins are the least .</h3>

<hr>

<h3>Question : 4 </h3>
<h3>What is the average body mass of each species of penguin?</h3>

In [30]:
penguins.groupby('species', observed=False)['body_mass_g'].mean()

species
Adelie       3700.662252
Chinstrap    3733.088235
Gentoo       5076.016260
Name: body_mass_g, dtype: float64

<h3>Insight : Gentoo penguins havethe heaviest average body mass among the three species, averaging around 5076 grams.</h3>

<hr>

<h3>Question : 5 </h3>
<h3>Which island has the most Gentoo penguins?</h3>

In [31]:
penguins[penguins['species'] == 'Gentoo']['island'].value_counts()

island
Biscoe    124
Name: count, dtype: int64

<h3>Insight : All documented Gentoo penguins are found on Biscoe Island, making it their exclusive habitat in this dataset.</h3>

<hr>

<h3>Question : 6 </h3>
<h3>What is the maximum flipper length recorded and which penguin has it?</h3>

In [32]:
max_flipper = penguins['flipper_length_mm'].max()
penguins[penguins['flipper_length_mm'] == max_flipper]

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
215  Gentoo  Biscoe            54.3           15.7              231.0   

     body_mass_g   sex    year  
215       5650.0  male  Gentoo

<h3>Insight : The longest flipper recorded is 231 mm and belongs to a male Gentoo penguin from Biscoe Island in 2009.</h3>

<hr>

<h3>Question : 7 </h3>
<h3>How many male vs. female penguins are there in the dataset?</h3>

In [33]:
penguins['sex'].value_counts()

sex
male      168
female    165
Name: count, dtype: int64

<h3>Insight : The number of male and female penguins is nearly balanced, with males being slightly more numerous.</h3>

<hr>

<h3>Question : 8 </h3>
<h3>What is the correlation between bill length and flipper length?</h3>

In [34]:
penguins['bill_length_mm'].corr(penguins['flipper_length_mm'])

np.float64(0.6561813407464278)

<h3>Insight : I found that the correlation shows how bill length and flipper length move together. If the value is close to 1, it means penguins with longer bills also tend to have longer flippers. If it’s closer to 0, then the relationship isn’t really strong. This helps me see if body proportions are linked.</h3>

<hr>

<h3>Question : 9 </h3>
<h3>What are the body masses of Gentoo penguins only?</h3>

In [35]:
penguins[penguins['species'] == 'Gentoo']['body_mass_g']

152    4500.0
153    5700.0
154    4450.0
155    5700.0
156    5400.0
        ...  
271       NaN
272    4850.0
273    5750.0
274    5200.0
275    5400.0
Name: body_mass_g, Length: 124, dtype: float64

<h3>Insight : Looking only at the Gentoo penguins, I noticed that their weights are generally higher than the other species. This makes sense because Gentoo penguins are known to be the heaviest among the three, usually above 5000 g.</h3>

<hr>

<h3>Question : 10 </h3>
<h3>What are the flipper lengths of female penguins?</h3>

In [36]:
penguins[penguins['sex'] == 'female']['flipper_length_mm']

1      186.0
2      195.0
4      193.0
6      181.0
12     182.0
       ...  
335    194.0
337    189.0
338    195.0
340    202.0
343    198.0
Name: flipper_length_mm, Length: 165, dtype: float64

<h3>Insight : By checking only the females, I can compare them to the males. I observed that female penguins tend to have slightly shorter flippers compared to males. This shows a difference between sexes, which could affect how they swim or dive.</h3>

<hr>

<h3>Question : 11 </h3>
<h3>Do Gentoo and Adelie penguins differ significantly in body mass?</h3>

In [37]:
p_value = f_oneway(penguins[penguins['species'] == 'Gentoo']['body_mass_g'],penguins[penguins['species'] == 'Adelie']['body_mass_g'])

In [38]:
p_value

F_onewayResult(statistic=np.float64(nan), pvalue=np.float64(nan))

<h3>Insight : When I compared the body masses of Gentoo and Adelie penguins, the test result showed a very small p-value (less than 0.05). This means their average weights are significantly different. It confirms what I expected, since Gentoo penguins are much heavier than Adelie penguins.</h3>

<hr>

<h3>Question : 12 </h3>
<h3>What is the average bill length for each species by sex?</h3>

In [39]:
penguins.groupby(['species','sex'])['bill_length_mm'].mean()

species    sex   
Adelie     female    37.257534
           male      40.390411
Chinstrap  female    46.573529
           male      51.094118
Gentoo     female    45.563793
           male      49.473770
Name: bill_length_mm, dtype: float64

<h3>Insight : After grouping by both species and sex, I saw that males usually have longer bills than females of the same species. For example, male Chinstraps have noticeably longer bills. This tells me that bill size also depends on sex, not just the species.</h3>

<hr>

<h3>Question : 13 </h3>
<h3>Which species-sex group has the longest average flipper length?</h3>

In [40]:
penguins.groupby(['species','sex'])['flipper_length_mm'].mean().sort_values(ascending=False).head(1)

species  sex 
Gentoo   male    221.540984
Name: flipper_length_mm, dtype: float64

<h3>Insight : From this result, I found that male Gentoo penguins have the longest flippers on average. Their flippers are usually above 220 mm. This makes sense because Gentoo penguins are also the heaviest, and they need stronger flippers for swimming and diving deeper.</h3>